bats/sql-create-database.bats

```diff
@@ -0,0 +1,164 @@
+load $BATS_TEST_DIRNAME/helper/common.bash
+
+setup() {
+    setup_common
+}
+
+teardown() {
+    teardown_common
+}
+
+@test "sql create new database" {
+    run dolt sql << SQL
+CREATE DATABASE mydb;
+SHOW DATABASES;
+USE mydb;
+CREATE TABLE test (
+    pk int primary key
+);
+INSERT INTO test VALUES (222);
+SELECT COUNT(*) FROM test WHERE pk=222;
+DROP DATABASE mydb;
+SQL
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "dolt_repo_$$" ]] || false
+    [[ "$output" =~ "information_schema" ]] || false
+    [[ "$output" =~ "mydb" ]] || false
+    # From COUNT
+    [[ "$output" =~ "1" ]] || false
+
+    run dolt sql -q "SHOW DATABASES"
+    [[ ! "$output" =~ "mydb" ]] || false
+}
+
+@test "sql create database that already exists throws an error" {
+    run dolt sql << SQL
+CREATE DATABASE mydb;
+CREATE DATABASE mydb;
+SQL
+
+    [ "$status" -eq 1 ]
+    [[ "$output" =~ "can't create database mydb; database exists" ]] || false
+
+}
+
+@test "sql create database IF NOT EXISTS on database that already exists doesn't throw an error" {
+    run dolt sql << SQL
+CREATE DATABASE mydb;
+CREATE DATABASE IF NOT EXISTS mydb;
+SQL
+
+    [ "$status" -eq 0 ]
+}
+
+@test "sql create and drop new database" {
+    run dolt sql << SQL
+CREATE DATABASE mydb;
+DROP DATABASE mydb;
+USE mydb;
+SQL
```

Comment:
This is an interesting pattern. Usually, I run these all as separate `dolt sql` commands and then test the one I want output for using `run`. I'm worried that running these all as on SQL block makes them harder to debug. So imagine we broke `DROP` in this case. You wouldn't be sure which one of those three statements broke. Make sense?

Reply:
Yup I usually do that but these in memory dbs expire on every session. So this was the best way to test besides the server.